drop-transfer/src/ws/server/mod.rs

```diff
@@ -938,9 +958,9 @@ impl<'a> FileStreamCtx<'a> {
             } = self;
 
             jobs.spawn(async move {
-                let _guard = guard;
+                let _gurad = guard.clone();
 
-                task.run(state, events, downloader, stream, req_send, logger)
+                task.run(state, events, downloader, stream, req_send, logger, guard)
```

Comment:
why do we clone the guard and not use it? also it contains a typo

Reply:
The copy is passed in because the function `run()` can spawn its own tasks, which we want to have control over when they finish.
I'll fix the typo :+1:

Reply:
Typo fixed

Reply:
but I mean why do we even need the `_guard` if we don't use it - do we gain something from making a copy of guard and dropping it almost immediately?

Reply:
It's not dropped immediately. `run()` function is async and the `_guard` is dropped only after `run()` finishes the execution

Reply:
what would happen if that line would be deleted?

Reply:
Which one?
If you meant this one `let _gurad = guard.clone();` then there is a possibility of shutting down the tokio runtime in `norddrop_close()` before the task finishes. We've seen this before, it leads to strange behaviors like random panics and so on.
In fact, the guards were introduced just to cope with this problem

Reply:
The `norddrop_close()` waits for all the guards to be dropped and only then return to the caller

Reply:
thanks for explaining